src/utils/restrict.ts

```diff
@@ -0,0 +1,27 @@
+import cheerio, { CheerioAPI } from "cheerio";
+
+import axios from "axios";
+
+interface CheckRestricted extends CheerioAPI {
+  kind: "CheckRestricted";
+}
+
+export interface Restricted {
+ locked: boolean;
+}
+
+export const getCheckingURL = (workID: string) => 
```

Comment:
Makes sense! Should this be shelved until that's established then? Or am I just not thinking of where to go to get the relevant information 

My only qualm is that making it modular seems like it would make things simpler if someone *just* wants to know restricted y/n status for some reason, but I'm probably overthinking it?

Reply:
> Makes sense! Should this be shelved until that's established then?
Since I don't think this work is being done in any of the outstanding PRs yet, I would simply put it in the right place and import it.

> My only qualm is that making it modular seems like it would make things simpler if someone just wants to know restricted y/n status for some reason, but I'm probably overthinking it?

I would say this is a consideration for later (but also there's a bunch of caveats I don't think we should discuss here). FWIW, I'd suggest extracting all the URLs methods in their own file.

Reply:
Should I do the second fix and then close this pull request for later?